ion-schema/src/isl/isl_constraint.rs

```diff
@@ -658,7 +709,22 @@ impl IslConstraintImpl {
                         "fields constraint can not be empty",
                     ));
                 }
-                Ok(IslConstraintImpl::Fields(fields))
+                match isl_version {
+                    IslVersion::V1_0 => Ok(IslConstraintImpl::Fields(fields, false)),
+                    IslVersion::V2_0 => {
+                        if value.annotations().count() > 1
+                            || value.annotations().any(|a| a.text() != Some("closed"))
+                        {
+                            return Err(invalid_schema_error_raw(
+                                "fields constraint may only be annotated with 'closed'",
+                            ));
```

Comment:
Slightly more concise:
```suggestion
                            return invalid_schema_error("fields constraint may only be annotated with 'closed'");
```